### ISSUE TITLE
Support for GZip and Deflate compression on HTTP Requests

### ISSUE THREAD

**Issue by Badgerati** [Badgerati/Pode#504]
### Describe the Feature
After #502 I thought I'd look into compression in Requests a bit more, see if Pode could do it. Well the good news is, it can!

There's an initial commit to follow, but the idea is to add in support for GZip and Deflate compression on HTTP Requests (*not* Responses, that will be a separate issue).

If a request comes in, and it's using compression, the payload will be decompressed first before parsing for JSON/etc. in the normal manor.

On Pode type web servers this will work like any other server:
* Compress some JSON payload using GZip
* Send in a request with the headers: `Transfer-Encoding: gzip` and `Content-Type: application/json`

Servers that use the HttpListener default logic are a little weirder. HttpListener doesn't support compression and fails with a 501 using the `Transfer-Encoding` header , so Pode has to use a slightly different header instead:
* Compress some JSON payload using GZip
* Send in a request with the headers: `X-Transfer-Encoding: gzip` and `Content-Type: application/json`

^ Same steps also work for `deflate`.

Notes:
* Possibly add a new `-TransferEncoding` onto `Add-PodeRoute` to force a compression encoding.

**Comment by Badgerati** [comment]
This also adds in chunked encoding support to the Pode server type now.
